models/stock_picking.py

```diff
@@ -35,6 +35,32 @@ def _get_package_search_domain(self, package):
                 '|', ('move_line_ids.result_package_id', 'child_of', package.id),
                      ('move_line_ids.u_result_parent_package_id', '=', package.id)]
 
+    def is_compatible_package(self, package_name):
+        """ The package with name package_name is compatible
+            with the picking in self if:
+            - The package does not exist
+            - The package is not in stock
+            - The package is not used in any other picking
+        """
+        Picking = self.env['stock.picking']
+        Package = self.env['stock.quant.package']
+
+        self.ensure_one()
+        self.assert_valid_state()
+
+        res = True
+        pickings = Picking.get_pickings(package_name=package_name)
```

Comment:
Won't this find completed pickings as it's not setting which states to consider?

Reply:
It is fine, since it is not compatible if it has been used before in a different picking. I have updated the descriptions.